python/tank/util/metrics.py

```diff
@@ -32,6 +33,139 @@
 json = shotgun_api3.shotgun.json
 
 
+###############################################################################
+
+class PlatformInfo(object):
+    """
+    Metric utility class providing basic platform information
+    to individual emitted metric
+
+    Platform Official Documentation
+    https://docs.python.org/2/library/platform.html
+    """
+
+    __cached_platform_info = None
+
+    @classmethod
+    def get_darwin_version(cls):
+        """
+        Returns a macOS / OSX friendly version string such as:
+            10.7, 10.11, 10.12, etc
+
+        :return: A str of a simple OS version string.
+        """
+
+        os_version = "Unknown"
+
+        # Now that we have 'raw' output secured, try limiting the version
+        try:
+            raw_version_str = platform.mac_ver()[0]
+            os_version = raw_version_str
+
+            # Now that we have 'raw' output secured, we generalize the actual version.
+            # We do want to limit the number of possible OS version variant we get metrics for.
+            #
+            # For macOS / OSX we keep only the Major.minor
+            os_version = re.findall(r"\d*\.\d*", raw_version_str)[0]
+
+        except:
+            pass
+
+        return os_version
+
+    @classmethod
+    def get_linux_version(cls):
+        """
+        Returns a Linux friendly version string such as:
+            "Ubuntu 12", "Fedora 24", "Red Hat 7", "Debian 8" etc
+
+        :return: A str of a simple OS version string.
+        """
+        os_version = "Unknown"
+
+        try:
+            # Get the distributon name and capitalize word(s) (e.g.: Ubuntu, Red Hat)
+            distro = platform.linux_distribution()[0].title()
+            raw_version_str = platform.linux_distribution()[1]
+
+            # For Linux we really just want the 'major' version component
+            major_version_str = re.findall(r"\d*", raw_version_str)[0]
+            os_version = "%s %s" % (distro, major_version_str)
+
+        except:
+            pass
+
+        return os_version
+
+    @classmethod
+    def get_windows_version(cls):
+        """
+        Returns a Windows friendly version string such as:
+            2000, XP, 7, 10 etc.
+
+        :return: A str of a simple OS version string.
+        """
+        os_version = "Unknown"
+
+        try:
+            # On Windows, we can simply use the 'Release()' method
+            # as it returns a friendly name e.g: XP, 7, 10 etc.
+            os_version = platform.release()
+
+        except:
+            pass
+
+        return os_version
+
+    @classmethod
+    def get_platform_info(cls):
+        """
+        Returns a simple OS and OS version information of the underlying host.
+        The information is cached to to saves on subsequent calls.
+
+        :return: A dict of basic platform information such as:
+            {'OS Version': 'Debian 7', 'OS': 'Linux'}
```

Comment:
minor: this makes it seem like the return value is a dict of dicts. suggest just listing one example or rephrasing to say that here is a list of examples of possible return values.